src/services/item/plugins/action/service.ts

```diff
@@ -54,11 +56,18 @@ export class ActionItemService {
     filters: { view?: Context; sampleSize?: number },
   ): Promise<Action[]> {
     const { view = Context.Builder, sampleSize = DEFAULT_ACTIONS_SAMPLE_SIZE } = filters;
+
     // get item
     const item = await this.itemService.get(actor, repositories, itemId);
 
-    // check member has admin membership over item
-    await validatePermission(repositories, PermissionLevel.Admin, actor, item);
+    // check permission
+    const permission = actor
+      ? (await repositories.itemMembershipRepository.getInherited(item, actor, true))?.permission
+      : null;
```

Comment:
If you check `actor` beforehand, you don't need to do a "complex" conditional, I think it would be clearer.